scipy/stats/_morestats.py

```diff
@@ -4785,19 +4791,22 @@ def directional_stats(samples, *, axis=0, normalize=True):
     >>> 1 - dirstats.mean_resultant_length
     0.13397459716167093
     """
-    samples = np.asarray(samples)
+    xp = array_namespace(samples)
+    samples = xp.asarray(samples)
+    
     if samples.ndim < 2:
         raise ValueError("samples must at least be two-dimensional. "
                          f"Instead samples has shape: {samples.shape!r}")
-    samples = np.moveaxis(samples, axis, 0)
+    samples = xp.moveaxis(samples, axis, 0)
```

Comment:
Ooh I guess we can do this now? That's nice.

Reply:
I'll have a look at removing our temporary shims in a follow-up